### PR TITLE
Scores explore alr

### PR DESCRIPTION
**Pull request by reid-andrew**
## Description
Adds in functionality for Scores Explore page for story #9 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
- Adds nav bar to all pages with links to "Dashboard" and "Explore"
- Explore page shows scores from contributors other than logged-in user
- Adds basic login functionality for testing & seeds database
- Slight refactor of collaboration functionality & additional testing
- Adds additional fixture file for getting scores from a user with multiple scores
- Refactors get_scores method on Score poro to allow for getting scores from other users. When not passed a specific user id, method defaults to 'me' in order to preserve existing tests/functionality.

## RSpec Results
```
17 examples, 0 failures
297 / 298 LOC (99.66%) covered.
```

### CODE REVIEW

**Comment by reid-andrew** [comment]
My thoughts on the merge conflicts: 

- Sessions Controller - probably best to maintain code in Master. I'd hard-coded a value in for my own personal testing purposes.
- Sessions new.html.erb - Master is closer to where we'll eventually need to be, I had a button-only login just to make testing easier. Better to preserve what's in Master.
- Seeds - Can probably do a bit of both, ideally can add in flat ID & username for Zach's user.
- Collab_spec - I think this is just a case of an extra line of space, so can just clean that up easily.

**Comment by reid-andrew** [comment]
Attempted to add comment & accidentally closed. Please ignore closure & refer to earlier comment for my thoughts on merge conflicts.

File: app/facades/explore_facade.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class ExploreFacade
+  attr_reader :scores
+
+  def initialize(current_user)
+    @users =  User.where.not(id: current_user.id).where.not(flat_id: nil) #Need to make this pull location sorted list.
+    @scores = get_scores
+  end
```
Comment (reid-andrew) [inline app/facades/explore_facade.rb:7]: Deliberately trying to pass users other than current user who have a flat_id. This'll need replaced with a list of users from the microservice eventually.

File: app/facades/explore_facade.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  def get_scores
+    all_scores = []
+    @users.each do |user|
+      all_scores << Score.create(user.flat_id)
+    end
+    all_scores.flatten
+  end
+end
```
Comment (reid-andrew) [inline app/facades/explore_facade.rb:16]: This is going to make seperate API calls for each user. I think that's probably unavoidable, but we may want to explore an alternative. This is working, though.

File: config/routes.rb [hunk 3]
```diff
@@ -1,8 +1,14 @@
 Rails.application.routes.draw do
+  root to: 'sessions#new'
+
   namespace :users do
     resources :dashboard, only: [:index]
+    resources :explore, only: [:index]
```
Comment (zachholcomb) [inline config/routes.rb:6]: nice restful solution

File: app/facades/explore_facade.rb (hunk 1 above)
Comment (zachholcomb) [reply app/facades/explore_facade.rb:7]: Right that should be an easy switch over when we're ready

**Pull request merged**